modules/docs/mdx/5.0-MIGRATION-GUIDE.mdx

```diff
@@ -91,6 +92,279 @@ for alpha components and Preview as a space for beta components.
+
+##### Link Variant Migration
+
+To migrate, please use the `Hyperlink` component instead. Detailed usage information is available in
+the [variants section](#variants).
+
+#### Type Updates In-Depth Overview
+
+The new type tokens introduce a few major changes:
+
+- Introducing rem units
+- Creating new `type` object structure
+- Adding new type properties (`fontFamilies`, `fontSizes`, and `fontWeights`)
+- Updating and replacing variants
+
+#### Introducing Rem Units
+
+The new type hierarchy uses `rem` units instead of `px`. This update follows the guidance
+[from the WCAG spec](https://www.w3.org/TR/WCAG21/#resize-text) and provides better support for
+users who rely on zooming. If you'd like to learn more about `rem` and relative units, you can
+review this
+[documentation](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Values_and_units#ems_and_rems).
+
+> **Note:** We are using `16px` as our base font-size for these values. This is a browser standard
+> and also fairly common across Workday. However, _if your body text is set to a value other than
+> `16px`_, you will need to adjust that value for text to render properly.
+
+#### New Type Object Structure
+
+The `beta` and `legacy` type object structures were fairly flat, provided many levels in the type
+hierarchy, and quite a few variants. While none of these are bad attributes, we found in our
+research that they created a large amount of confusion. Both designers and engineers were unclear on
+when to use many of the tokens provided. We restructured the object to help users make more sense of
+it. The tokens are divided into three main parts:
+
+- `levels` (the type hierarchy)
+- `properties` (`fontFamilies`, `fontSizes`, and `fontWeights`)
+- `variants` (modifiers for type styles)
+
+##### Levels
+
+Type `levels` contain our new type hierarchy. When applying type styles, we recommend using these
+tokens first. Each size applies `fontFamily`, `fontSize`, `fontWeight`, `lineHeight`,
+`letterSpacing`, and `color` styles for you, so you can create consistent type quickly and easily.
+Instead of the previous flat structure, the type hierarchy is now organized in five levels:
+
+- `display` (used in large heroes and images, not typically used in product applications)
+- `title` (used for large page titles)
+- `heading` (used for headings and large text)
+- `body` (used for standard body text)
+- `subtext` (used for small subtext content or in tight spaces)
+
+And each level has three sizes: `large`, `medium`, and `small`. This organization allows the
+hierarchy to be more flexible and provide better guidance for usage. Below is an example:
+
+```tsx
+// v4
+import {type} from '@workday/canvas-kit-labs-react-core';
+
+const headingStyles = type.h2;
+const smallBodyStyles = type.body2;
+
+// v5
+import {type} from '@workday/canvas-kit-react/tokens';
+
+const headingStyles = type.levels.heading.large;
+const smallBodyStyles = type.levels.subtext.large;
+```
```

Comment:
I would also call out that this addresses the awkward issue from our old hierarchy where you would apply a stylistic heading level to a different level of semantic element (e.g. `<h2 css={ type.h4 }>`)

Reply:
Yeah, that's what I meant in the sentence above:
> This organization allows the hierarchy to be more flexible and provide better guidance for usage.

But I can add an example as well.

Reply:
Done.